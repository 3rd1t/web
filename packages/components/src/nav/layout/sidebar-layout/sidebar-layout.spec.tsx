import React from "react"
import { render } from "@testing-library/react"

import MultiColumn from "./multi-column"

describe("MultiColumn", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<MultiColumn />)
    expect(baseElement).toBeTruthy()
  })
})
