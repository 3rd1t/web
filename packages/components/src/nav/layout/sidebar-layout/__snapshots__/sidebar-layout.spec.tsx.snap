// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLayout primary should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex"
    >
      <div
        class="sticky top-0 flex flex-shrink-0 min-h-screen"
      >
        <div
          style="width: 320px;"
        >
          <button
            class="text-white"
          >
            Toggle
          </button>
          <div
            class="relative"
          >
            <div
              class="absolute border-r border-gray-300"
            >
              <nav
                class="relative flex flex-col justify-between min-h-screen pt-12 bg-gray-200"
                style="width: 80px;"
              >
                <div>
                  <a
                    class="flex items-center w-12 h-12 m-4 focus:outline-none"
                    href="/"
                  >
                    <div
                      class="flex items-center space-x-2"
                    >
                      <svg
                        class="w-full stroke-current text-data-500"
                        fill="none"
                        viewBox="0 0 194 148"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9 106.208L97.3883 17.8199L134.158 54.5894M185.07 41.8615L96.6814 130.25L59.9118 93.4803"
                          stroke-width="25"
                        />
                      </svg>
                    </div>
                  </a>
                  <div
                    class="mt-12"
                  >
                    <a
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:bg-data-400 hover:text-white cursor-pointer"
                      href="/factor/3"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6 trending-up"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M13 7h8m0 0v8m0-8l-8 8-4-4-6 6"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </a>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6 globe-alt"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M21 12a9 9 0 01-9 9m9-9a9 9 0 00-9-9m9 9H3m9 9a9 9 0 01-9-9m9 9c1.657 0 3-4.03 3-9s-1.343-9-3-9m0 18c-1.657 0-3-4.03-3-9s1.343-9 3-9m-9 9a9 9 0 019-9"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6 chip"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M9 3v2m6-2v2M9 19v2m6-2v2M5 9H3m2 6H3m18-6h-2m2 6h-2M7 19h10a2 2 0 002-2V7a2 2 0 00-2-2H7a2 2 0 00-2 2v10a2 2 0 002 2zM9 9h6v6H9V9z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6 currency-dollar"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c-1.657 0-3 .895-3 2s1.343 2 3 2 3 .895 3 2-1.343 2-3 2m0-8c1.11 0 2.08.402 2.599 1M12 8V7m0 1v8m0 0v1m0-1c-1.11 0-2.08-.402-2.599-1M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6 library"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8 14v3m4-3v3m4-3v3M3 21h18M3 10h18M3 7l9-4 9 4M4 10h16v11H4V10z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:bg-data-400 hover:text-white cursor-pointer"
                  >
                    <span
                      class="flex items-center justify-center w-12"
                    >
                      <svg
                        class="w-6 h-6"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M14 5l7 7m0 0l-7 7m7-7H3"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                  >
                    <span
                      class="flex items-center justify-center w-12"
                    >
                      <svg
                        class="w-6 h-6"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </nav>
            </div>
            <div
              class=""
            >
              <div
                class="flex flex-col justify-between min-h-screen px-4 py-4 ml-20 bg-white border-r border-gray-300"
              >
                <ul
                  class="flex flex-col w-full"
                >
                  <li>
                    <span
                      class="flex my-4 text-sm font-medium text-gray-800 uppercase hover:text-gray-600"
                    >
                      Builder
                    </span>
                  </li>
                  <li>
                    <a
                      aria-label="Settings"
                      class="flex items-center h-12 space-x-2 text-gray-400 rounded-lg hover:text-gray-900"
                      href="/notImplemented"
                    >
                      <svg
                        class="w-6 h-6 code"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 20l4-16m4 4l4 4-4 4M6 16l-4-4 4-4"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                      <div
                        class="flex items-center justify-between space-x-1"
                      >
                        <span
                          class="text-gray-700 hover:text-gray-900"
                        >
                          API
                        </span>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      aria-label="Settings"
                      class="flex items-center h-12 space-x-2 text-gray-400 rounded-lg hover:text-gray-900"
                      href="/notImplemented"
                    >
                      <svg
                        class="w-6 h-6 cloud-download"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M9 19l3 3m0 0l3-3m-3 3V10"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                      <div
                        class="flex items-center justify-between space-x-1"
                      >
                        <span
                          class="text-gray-700 hover:text-gray-900"
                        >
                          CSV Download
                        </span>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      aria-label="Settings"
                      class="flex items-center h-12 space-x-2 text-gray-400 rounded-lg hover:text-gray-900"
                      href="/notImplemented"
                    >
                      <svg
                        class="w-6 h-6 chart-square-bar"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16 8v8m-4-5v5m-4-2v2m-2 4h12a2 2 0 002-2V6a2 2 0 00-2-2H6a2 2 0 00-2 2v12a2 2 0 002 2z"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                      <div
                        class="flex items-center justify-between space-x-1"
                      >
                        <span
                          class="text-gray-700 hover:text-gray-900"
                        >
                          Charts
                        </span>
                        <span
                          class="flex items-center justify-center h-6 px-2 text-xs font-semibold uppercase bg-opacity-25 rounded bg-data-400 text-data-700"
                        >
                          coming soon
                        </span>
                      </div>
                    </a>
                  </li>
                  <li
                    class="mt-8"
                  >
                    <span
                      class="flex my-4 text-sm font-medium text-gray-800 uppercase hover:text-gray-600"
                    >
                      Factor models
                    </span>
                  </li>
                  <li>
                    <a
                      class="flex items-start h-16 hover:text-gray-900 "
                      href="/notImplemented"
                    >
                      <span
                        class="text-gray-700"
                      >
                        Three
                      </span>
                      <div
                        class="flex flex-col items-end w-full"
                      >
                        <span
                          class="text-sm text-gray-700 hover:text-gray-900"
                        >
                          Fama & French
                        </span>
                        <span
                          class="flex items-center justify-center text-xs font-thin text-gray-700"
                        >
                          1993
                        </span>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      class="flex items-start h-16 hover:text-gray-900 "
                      href="/notImplemented"
                    >
                      <span
                        class="text-gray-700"
                      >
                        Four
                      </span>
                      <div
                        class="flex flex-col items-end w-full"
                      >
                        <span
                          class="text-sm text-gray-700 hover:text-gray-900"
                        >
                          Carhart
                        </span>
                        <span
                          class="flex items-center justify-center text-xs font-thin text-gray-700"
                        >
                          1997
                        </span>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      class="flex items-start h-16 hover:text-gray-900 "
                      href="/notImplemented"
                    >
                      <span
                        class="text-gray-700"
                      >
                        Five
                      </span>
                      <div
                        class="flex flex-col items-end w-full"
                      >
                        <span
                          class="text-sm text-gray-700 hover:text-gray-900"
                        >
                          Fama & French
                        </span>
                        <span
                          class="flex items-center justify-center text-xs font-thin text-gray-700"
                        >
                          2015
                        </span>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      class="flex items-start h-16 hover:text-gray-900 "
                      href="/notImplemented"
                    >
                      <span
                        class="text-gray-700"
                      >
                        Six
                      </span>
                      <div
                        class="flex flex-col items-end w-full"
                      >
                        <span
                          class="text-sm text-gray-700 hover:text-gray-900"
                        >
                          Fama & French
                        </span>
                        <span
                          class="flex items-center justify-center text-xs font-thin text-gray-700"
                        >
                          2018
                        </span>
                      </div>
                    </a>
                  </li>
                </ul>
                <div>
                  <div
                    class="pt-3 border-t border-gray-300"
                  >
                    <a
                      aria-label="Settings"
                      class="flex items-center h-12 space-x-2 text-gray-400 rounded-lg hover:text-gray-900"
                      href="mailto:request@perfol.io"
                    >
                      <svg
                        class="w-6 h-6 text-data-400"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                      <div
                        class="flex items-center justify-between space-x-1"
                      >
                        <span
                          class="text-gray-700 hover:text-gray-900"
                        >
                          Request feature
                        </span>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full min-h-screen"
      >
        <div
          class="w-full h-20"
        >
          <ul
            class="flex items-center justify-between h-full px-4"
          >
            <li
              class="text-gray-700 "
            >
              <div
                class="flex items-center space-x-2"
              >
                <span>
                  <span
                    class="text-gray-700"
                  >
                    First
                  </span>
                </span>
                <span>
                  <span
                    class="font-thin text-gray-700"
                  >
                    /
                  </span>
                </span>
                <span>
                  <span
                    class="text-gray-700"
                  >
                    Second
                  </span>
                </span>
                <span>
                  <span
                    class="font-thin text-gray-700"
                  >
                    /
                  </span>
                </span>
                <span>
                  <span
                    class="font-medium text-data-700"
                  >
                    Active
                  </span>
                </span>
              </div>
            </li>
            <li>
              <ul
                class="flex items-center justify-between space-x-16"
              >
                <li>
                  <a
                    href="/docs"
                  >
                    <div
                      class="focus:outline-none transition duration-200 text-gray-800 hover:text-gray-700   "
                    >
                      <div
                        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
                      >
                        <span>
                          Documentation
                        </span>
                      </div>
                    </div>
                  </a>
                </li>
                <li>
                  <div
                    class="relative"
                  >
                    <button
                      aria-haspopup="true"
                      class="focus:outline-none"
                      id="headlessui-menu-button-1"
                      type="button"
                    >
                      <div
                        class="flex flex-col items-end w-40 rounded hover:text-data-800 hover:font-semibold"
                      >
                        <span>
                          Vin Venture
                        </span>
                        <span
                          class="text-xs font-semibold text-data-700"
                        >
                          Premium
                        </span>
                      </div>
                    </button>
                  </div>
                </li>
              </ul>
            </li>
          </ul>
        </div>
        <main
          class="h-full pt-20 -mt-20 "
        >
          <div
            class="h-full bg-gray-100 border-t border-gray-300"
          >
            <span>
              Content
            </span>
          </div>
        </main>
      </div>
    </div>
  </div>
</body>
`;
